cached_parser.go

```diff
@@ -0,0 +1,88 @@
+package editorconfig
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"gopkg.in/ini.v1"
+)
+
+type CachedParser struct {
+	iniFiles map[string]*ini.File
+	regexps  map[string]*regexp.Regexp
+	metrics  map[string]int
+}
+
+func NewCachedParser() *CachedParser {
+	metrics := make(map[string]int)
+
+	metrics["ini_hit"] = 0
+	metrics["ini_miss"] = 0
+	metrics["fnmatch_hit"] = 0
+	metrics["fnmatch_miss"] = 0
```

Comment:
```go
metrics := map[string]int{
	"ini_hit": 0,
	"ini_miss": 0,
	"fnmatch_hit": 0,
	"fnmatch_miss": 0,
}
```

Reply:
Maybe this could be a struct, actually (since it only has these specific 4 fields)

Reply:
I'll drop the metrics for now.